numcosmo/lss/nc_galaxy_wl_ellipticity_kde.c

```diff
@@ -0,0 +1,367 @@
+ * @stability: Unstable
+ *
+ *
+ * Class defining a galaxy weak lensing ellipticity with Kernel Density Estimation.
+ * probability distribution $P_\mathrm{wl}(g)$.
+ *
+ */
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif /* HAVE_CONFIG_H */
+#include "build_cfg.h"
+
+#include "lss/nc_galaxy_wl_ellipticity_kde.h"
+#include "nc_enum_types.h"
+#include "math/ncm_stats_dist1d_epdf.h"
+#include "lss/nc_galaxy_redshift_spec.h"
+
+#include <math.h>
+#include <gsl/gsl_math.h>
+
+struct _NcGalaxyWLEllipticityKDEPrivate
+{
+  NcmMatrix *obs;
+  NcmStatsDist1d *kde;
+  NcmVector *e_vec;
+  gdouble r;
+  gdouble twolnN;
+  guint len;
+};
+
+enum
+{
+  PROP_0,
+  PROP_OBS,
+};
+
+G_DEFINE_TYPE_WITH_PRIVATE (NcGalaxyWLEllipticityKDE, nc_galaxy_wl_ellipticity_kde, NC_TYPE_GALAXY_WL_DIST);
+
+static void
+nc_galaxy_wl_ellipticity_kde_init (NcGalaxyWLEllipticityKDE *grsg)
+{
+  NcGalaxyWLEllipticityKDEPrivate * const self = grsg->priv = nc_galaxy_wl_ellipticity_kde_get_instance_private (grsg);
+  
+  self->obs    = NULL;
+  self->kde    = NULL;
+  self->e_vec  = NULL;
+  self->r      = 0.0;
+  self->twolnN = 0.0;
+  self->len    = 0;
+}
+
+static void
+_nc_galaxy_wl_ellipticity_kde_set_property (GObject *object, guint prop_id, const GValue *value, GParamSpec *pspec)
+{
+  NcGalaxyWLEllipticityKDE *grsg = NC_GALAXY_WL_ELLIPTICITY_KDE (object);
+  
+  g_return_if_fail (NC_IS_GALAXY_WL_ELLIPTICITY_KDE (object));
+  
+  switch (prop_id)
+  {
+    case PROP_OBS:
+      nc_galaxy_wl_ellipticity_kde_set_obs (grsg, g_value_get_object (value));
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+static void
+_nc_galaxy_wl_ellipticity_kde_get_property (GObject *object, guint prop_id, GValue *value, GParamSpec *pspec)
+{
+  NcGalaxyWLEllipticityKDE *grsg = NC_GALAXY_WL_ELLIPTICITY_KDE (object);
+  
+  g_return_if_fail (NC_IS_GALAXY_WL_ELLIPTICITY_KDE (object));
+  
+  switch (prop_id)
+  {
+    case PROP_OBS:
+      g_value_set_object (value, nc_galaxy_wl_ellipticity_kde_peek_obs (grsg));
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+static void
+_nc_galaxy_wl_ellipticity_kde_dispose (GObject *object)
+{
+  /* Chain up : end */
+  G_OBJECT_CLASS (nc_galaxy_wl_ellipticity_kde_parent_class)->dispose (object);
+}
+
+static void
+_nc_galaxy_wl_ellipticity_kde_finalize (GObject *object)
+{
+  /* Chain up : end */
+  G_OBJECT_CLASS (nc_galaxy_wl_ellipticity_kde_parent_class)->finalize (object);
+}
+
+static void _nc_galaxy_wl_ellipticity_kde_m2lnP_initial_prep (NcGalaxyWLDist *gwld,NcGalaxyRedshift *gz, NcHICosmo *cosmo, NcHaloDensityProfile *dp, NcWLSurfaceMassDensity *smd, const gdouble z_cluster);
+// static void _nc_galaxy_wl_ellipticity_kde_m2lnP_prep (NcGalaxyWLDist *gwld, NcHICosmo *cosmo, NcHaloDensityProfile *dp, NcWLSurfaceMassDensity *smd, const gdouble z_cluster, const guint gal_i);
```

Comment:
Why this method is commented all over the place?

Reply:
I thought this method was optional for these objects, and since I don't need it for the KDE, I simply removed it. Do we need it?